newlib/libc/tinystdio/fseek.c

```diff
@@ -54,3 +55,15 @@ int FSEEK(FILE *stream, FSEEK_TYPE offset, int whence)
 	errno = ESPIPE;
 	return -1;
 }
+
+#if defined(_WANT_FLOCKFILE)
+int
+FSEEK(FILE *stream, FSEEK_TYPE offset, int whence)
+{
+    int ret;
+    __flockfile(stream);
+    ret = FILE_FN_UNLOCKED(FSEEK)(stream, offset, whence);
```

Comment:
Hmm,

These all looks really suspicious for me.
I understand how the addition of locking around `printf` can fix https://github.com/picolibc/picolibc/issues/760 but IMO it doesn't make sense with most other functions. I.E. I can't call `fseek` / `fgetc` / etc... simultaneously on same file stream from different threads on different CPUs (example from https://github.com/picolibc/picolibc/issues/760) and expect any meaningful result.

Am I missing something?

Reply:
@evgeniy-paltsev , flockfile should lock all stream operations isn't it? 

When I created https://github.com/picolibc/picolibc/issues/760 I expected behavior like in popular libc implementations e.g.: [glibc](https://github.com/lattera/glibc/blob/master/libio/fseek.c) or [newlib](https://github.com/bminor/newlib/blob/master/newlib/libc/stdio/fseeko.c#L116)

Reply:
It might not be meaningful, but (at least) the buffered I/O layer needs to have locking for fseek to not scramble the internal state. As this global locking implementation disables the internal buffered I/O locking code, we'll need top-level locking here, I think.